modbusApp/src/modbusInterpose.c

```diff
@@ -247,8 +249,10 @@ static asynStatus writeIt(void *ppvt, asynUser *pasynUser,
     switch(pPvt->linkType) {
         case modbusLinkTCP:
             /* Build the MBAP header */
-            pPvt->transactionId = (pPvt->transactionId + 1) & 0xFFFF;
-            mbapHeader.transactId    = htons(pPvt->transactionId);
+            if (pPvt->skipTransactionId != 0) {
```

Comment:
I think this means that `mbapHeader.transactId` will be uninitialised when sent in no check mode, which could potentially cause debug checks on uninitisalised usage of variables to fail. Probabyl best to just leave the transaction id sent logic as before, but just nopt check teh one sent back

Reply:
makes sense - hopefully most recent commit just does the read check